extmod/asyncio/stream.py

```diff
@@ -114,6 +122,18 @@ def open_connection(host, port):
         if er.errno != EINPROGRESS:
             raise er
     yield core._io_queue.queue_write(s)
+    # wrap with SSL, if requested
+    if ssl:
+        if not _ssl:
+            raise ValueError("SSL not supported")
+        if ssl is True:
+            ssl = _ssl.SSLContext(_ssl.PROTOCOL_TLS_CLIENT)
```

Comment:
Instead of importing ssl at the top of this file, it can be delayed to this point.  Then this function can be used with a custom SSLContext passed in as the `ssl` argument and the port doesn't even need to have the `ssl` module.

I suggest:
```py
if ssl is True:
    import ssl as _ssl

    ssl = _ssl.SSLContext(_ssl.PROTOCOL_TLS_CLIENT)
```

If the `ssl` module doesn't exist then it'll raise an import error, which I think is good enough to inform the user that SSL is not supported because the `ssl` module does not exist.